packages/native/src/components/Button/getButtonStyle.ts

```diff
@@ -1,75 +1,140 @@
-            padding: `0 ${p.theme.space[4]}px`,
-          }
-        : {
-            borderColor: p.theme.colors.palette.neutral.c40,
-            padding: `0 ${p.theme.space[4]}px`,
-          };
+  const { type, disabled, pressed, outline } = props;
+
+  const { palette } = colors;
+
+  const primaryColor = palette.neutral.c00;
+  let secondaryColor;
+  let tertiaryColor;
+
+  const disabledColors = {
+    primaryColor: palette.neutral.c50,
+    secondaryColor: outline ? palette.neutral.c50 : palette.neutral.c30,
+  };
+
+  switch (type) {
+    case "main":
+      if (disabled) {
+        return disabledColors;
+      } else if (pressed && !outline) {
+        secondaryColor = palette.neutral.c80;
+      } else {
+        secondaryColor = palette.neutral.c100;
+        tertiaryColor = palette.neutral.c30;
+      }
+      break;
+    case "error":
+      if (disabled) {
+        return disabledColors;
+      } else if (pressed && !outline) {
+        secondaryColor = palette.error.c60;
+      } else {
+        secondaryColor = palette.error.c100;
+        tertiaryColor = palette.error.c30;
+      }
+      break;
+    case "color":
     default:
-      return {};
+      if (disabled) {
+        return disabledColors;
+      } else if (pressed && !outline) {
+        secondaryColor = palette.primary.c70;
+      } else {
+        secondaryColor = palette.primary.c80;
+        tertiaryColor = palette.primary.c20;
+      }
+      break;
   }
+
+  return {
+    primaryColor,
+    secondaryColor,
+    tertiaryColor,
+  };
 }
 
-export function getButtonColor(p: {
-  type?: ButtonTypes | undefined;
-  disabled?: boolean;
-  theme: Theme;
-}): {
-  color: string;
+export function getButtonColorStyle(
+  colors: Theme["colors"],
+  props: ButtonProps
+): {
+  text: {
+    color: string;
+  };
+  button: {
+    backgroundColor: string;
+    borderColor?: string;
+    borderWidth?: number;
+  };
 } {
-  switch (p.type) {
-    case "primary":
-      return p.disabled
-        ? {
-            color: p.theme.colors.palette.neutral.c70,
-          }
-        : {
-            color: p.theme.colors.palette.neutral.c00,
-          };
-    case "secondary":
-      return p.disabled
-        ? {
-            color: p.theme.colors.palette.neutral.c40,
-          }
-        : {
-            color: p.theme.colors.palette.neutral.c100,
-          };
+  const { outline, pressed } = props;
+  const { primaryColor, secondaryColor, tertiaryColor } = getButtonColors(
+    colors,
+    props
+  );
+
+  if (outline) {
+    return {
+      text: { color: secondaryColor },
+      button: {
+        backgroundColor:
+          pressed && tertiaryColor ? tertiaryColor : "transparent",
+        borderColor: secondaryColor,
+        borderWidth: 1,
+      },
+    };
+  } else {
+    return {
+      text: { color: primaryColor },
+      button: {
+        backgroundColor: secondaryColor,
+      },
+    };
+  }
+}
+
+export function getButtonSizeStyle({ size }: ButtonProps): {
```

Comment:
same issue with uneccesary swith for me here
since you only switch on one value you could do this without a function + a switch :shrug: 
```js
const buttonSizeStyle = {
   small: {
        padding: "0 20px",
        height: "40px",
      },
  ...
 }
buttonSizeStyle[size || "medium"]
```

Reply:
done